### ISSUE TITLE
Filters should auto-focus the text field

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-admin#1584]
**Steps to reproduce:**
- Go to the _Products_ report.
- In the filters drop down, select _Single Product_.
- Notice the button to go back is focused.

![filters-focus](https://user-images.githubusercontent.com/3616980/52861549-21b34280-3133-11e9-8c8c-b575613ee5f5.gif)

**Expected results:**
- The text field should be focused, so users can start typing right away.

**Comment by psealock** [comment]
### Prior Art

I have attempted this one and am facing some kind of odd race condition when the search component has a tag in it. Upon mounting the popover, the Tag's "remove" button will successfully receive focus 90% of the time in Chrome and none of the time in Firefox despite all the logic being wrapped in a `setTimeout` called in Popover's `componentDidMount`.

Branch here, https://github.com/woocommerce/woocommerce-admin/tree/fix/search-filter-focus

Accompanying Gutenberg changes here, https://github.com/WordPress/gutenberg/pull/14502

I'm going to pause on this in favour of working on features that move the product forward more for our users.

**Comment by timmyc** [comment]
@psealock I'm going to move this to the sprint 16 backlog - not sure if you are still actively working on this or not.

**Comment by psealock** [comment]
I'm not, but we can give it another shot in 16
